src/lib/application.ts

```diff
@@ -36,6 +42,84 @@ const supportedVersionMajorMinor = packageInfo.peerDependencies.typescript
     .split("||")
     .map((version) => version.replace(/^\s*|\.x\s*$/g, ""));
 
+/**
+ * Expand the provided packages configuration paths, determining the entry points
+ * and creating the ts.Programs for any which are found.
+ * @param logger
+ * @param packageGlobPaths
+ * @returns The information about the discovered programs, undefined if an error occurs.
+ */
+function getProgramsForPackages(
+    logger: Logger,
+    packageGlobPaths: string[]
+): PackageProgram[] | undefined {
+    const results = new Array<PackageProgram>();
+    // --packages arguments are workspace tree roots, or glob patterns
+    // This expands them to leave only leaf packages
+    const expandedPackages = expandPackages(logger, ".", packageGlobPaths);
+    for (const packagePath of expandedPackages) {
+        const packageJsonPath = resolve(packagePath, "package.json");
+        const packageJson = loadPackageManifest(logger, packageJsonPath);
+        if (packageJson === undefined) {
+            logger.error(`Could not load package manifest ${packageJsonPath}`);
+            return;
+        }
+        const packageEntryPoint = getTsEntryPointForPackage(
+            logger,
+            packageJsonPath,
+            packageJson
+        );
+        if (packageEntryPoint === undefined) {
```

Comment:
Question - why is failing to find an entry point sometimes a fatal error and sometimes just a warning? Is one reason for failure more common/expected?

Reply:
My razor here would be:
Ignore things which don't appear to be `require`-able node packages.
Fail on things which appear to be `require`-able node packages but are missing the necessary metadata for us to document.

I'll actually add the above as a comment in the code to guide maintainers.

The scenario which I saw so far, was just from running this across one of my own codebases where some of my packages were actually webpack browser bundles, not intended to be consumed as node.js modules, so there is no js entry point to be seen. This package for example: https://github.com/efokschaner/laniakea/tree/master/demos/balls/client

I fully anticipate we're going to see some surprising things in the wild w.r.t to packages and package structure for which this first pass at some default behaviour will not suffice. This comes back to my general remark above about potentially needing support for per-package typedoc config in the future, when we learn what people might need to tune.

Reply:
Makes sense :+1: 
Per-package configuration might be difficult... `excludeExternals` for example changes the theme output... future work.